### ISSUE TITLE
Is it possible to use a nswag.json for the settings when using OpenApiReference or OpenApiProjectReference 

### ISSUE THREAD

**Issue by redoz** [RicoSuter/NSwag#2955]
I've been poking around but I can't seem to figure it out, I'm guessing the only option is to use NSwag.MSbuild directly?

If it isn't possible, is it something that you would consider adding?

**Comment by RicoSuter** [comment]
It should work as far as i know - but i personally dont use it (i use NSwag.MSbuild) and the MS docs are quite sparse... 
The NSwag docs for that can be found here, please add links and update the wiki if you find out more: 
https://github.com/RicoSuter/NSwag/wiki/ServiceProjectReference

**Comment by redoz** [comment]
"Sparse" seems like an understatement :)

But I'm using `NSwag,MSbuild` with `run` and variables and that works for now. 

I did find out that the `OpenApiProjectReference` ends up calling nswag like this:
```xml
      <CurrentOpenApiReference>
        <Command>$(_NSwagCommand) swagger2tsclient /className:%(ClassName) /namespace:%(Namespace)</Command>
      </CurrentOpenApiReference>
      <CurrentOpenApiReference>
        <Command>%(Command) /input:"%(FullPath)" /output:"%(OutputPath)" %(Options)</Command>
      </CurrentOpenApiReference>
```
and based on what I've seen, you cannot use the `nswag.json` in conjunction with anything but the `run` command, is that correct?

**Comment by RicoSuter** [comment]
> and based on what I've seen, you cannot use the nswag.json in conjunction with anything but the run command, is that correct?

correct, the nswag.json is like a simple "pipeline" which does the openapi generation/loading and then passes it to the command

we'd need a way to specify a config file for the swagger2tsclient command (ie the sub object in the nswag.json config)

**Comment by frosadev** [comment]
Is this already an option?
I'm trying to generate a C# client but with some predefined options and, since I'm setting a lot of stuff, that could give a lot of help..

![image](https://user-images.githubusercontent.com/26621392/109324901-2f851d80-784d-11eb-826b-d05e4d141ce9.png)

**Comment by RicoSuter** [comment]
I think this calls directly into the generators and not via the nswag.json file... but i agree that it should be possible to provide a json file which is used instead of inline options
